src/Hl7.Fhir.Specification/Validation/TypeRefValidationExtensions.cs

```diff
@@ -32,46 +32,63 @@ internal static OperationOutcome ValidateType(this Validator validator, ElementD
             var choices = typeRefs.Select(tr => tr.Code).Distinct();
 
-            if (choices.Count() > 1)
+
+            if (instance.InstanceType != null)
             {
-                if (instance.InstanceType != null)
+                //find out what type is present in the instance data
+                // (e.g. deceased[Boolean], or _resourceType in json). This is exposed by IElementNavigator.TypeName.
+                var instanceType = ModelInfo.FhirTypeNameToFhirType(instance.InstanceType);
+
+                if (choices.Count() > 1)
                 {
-                    // This is a choice type, find out what type is present in the instance data
-                    // (e.g. deceased[Boolean], or _resourceType in json). This is exposed by IElementNavigator.TypeName.
-                    var instanceType = ModelInfo.FhirTypeNameToFhirType(instance.InstanceType);
-                    if (instanceType != null)
+                    if (instanceType is not null)
                     {
-                        // In fact, the next statements are just an optimalization, without them, we would do an ANY validation
+                        // The next statements are just an optimalization, without them, we would do an ANY validation
                         // against *all* choices, what we do here is pre-filtering for sensible choices, and report if there isn't
                         // any.
                         var applicableChoices = typeRefs.Where(tr => !tr.Code.StartsWith("http:"))
-                                        .Where(tr => ModelInfo.IsInstanceTypeFor(ModelInfo.FhirTypeNameToFhirType(tr.Code).Value,
-                                            instanceType.Value));
+                                    .Where(tr => ModelInfo.IsInstanceTypeFor(ModelInfo.FhirTypeNameToFhirType(tr.Code).Value, instanceType.Value));
 
                         // Instance typename must be one of the applicable types in the choice
                         if (applicableChoices.Any())
                         {
-                            outcome.Include(validator.ValidateTypeReferences(applicableChoices, instance, state));
+                            outcome.Include(validator.ValidateTypeReferences(applicableChoices, instance, state, validateProfiles));
                         }
                         else
                         {
                             var choiceList = String.Join(",", choices.Select(t => "'" + t + "'"));
                             validator.Trace(outcome, $"Type specified in the instance ('{instance.InstanceType}') is not one of the allowed choices ({choiceList})",
-                                     Issue.CONTENT_ELEMENT_HAS_INCORRECT_TYPE, instance);
+                                        Issue.CONTENT_ELEMENT_HAS_INCORRECT_TYPE, instance);
                         }
                     }
                     else
+                    {
```

Comment:
This block should move to line 41 (so just under `var instanceType = ModelIInfo.FhirTypeNameToFhirType(..)` : it's applicable to both choice and non-choice elements.